x-pack/legacy/plugins/siem/server/types.ts

```diff
@@ -5,31 +5,25 @@
  */
 
 import { Legacy } from 'kibana';
+import { XPackMainPlugin } from '../../xpack_main/xpack_main';
+
+export interface SiemSetupDependencies {
+  xpack: Pick<XPackMainPlugin, 'registerFeature'>;
+}
 
 export interface ServerFacade {
   config: Legacy.Server['config'];
```

Comment:
We currently only use this to fetch three external values: `pkg.version`, `xpack.alerting.enabled`, and `xpack.actions.enabled`. If I can find NP equivalents for these, the siem-specific config can be migrated to the new platform and consumed accordingly by the legacy plugin.

Reply:
You would declare them as depedencies, but the best way to test if they're enabled is to test if the dependency is defined. In general config values aren't exposed to other plugins unless they explicitly make their configuration available as an API.

```ts
class MyPlugin {
  public setup(core, plugins){
    if(plugins.alerting != null) {
      // you can use alerting
  }
}
```